InventoryManagementSystem/Domain/Repositories/MongoDbProductRepository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using MongoDB.Driver;
+using InventoryManagementSystem.Domain.Models;
+using InventoryManagementSystem.Domain.Repositories;
+
+namespace InventoryManagementSystem.DataAccess
+{
+    public class MongoDbProductRepository : IProductRepository
+    {
+        private readonly IMongoCollection<Product> _productCollection;
+
+        public MongoDbProductRepository()
+        {
+            var connectionString = "mongodb+srv://mahmoudbmiqdad:Mahmoud2003@cluster0.t1n1ssg.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
+            var client = new MongoClient(connectionString);  
+            var database = client.GetDatabase("InventoryDB"); 
+            _productCollection = database.GetCollection<Product>("products"); 
+        }
+
+        public void AddProduct(Product product)
+        {
+            try
+            {
+                _productCollection.InsertOne(product);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"An error occurred while adding the product: {ex.Message}");
```

Comment:
the repo shouldn't print things to screen, for this one you can just leave the exception be propagate to caller, and the caller handles the unexpected exceptions.

same comment for the other functions.